R/response_process.R

```diff
@@ -107,40 +113,38 @@ check_for_json <- function(resp, call = caller_env()) {
     return(invisible(resp))
   }
 
-  content <- httr::content(resp, as = "text")
   gargle_abort_request_failed(
-    c(
-      gargle_map_cli(
-        type,
-        template = "Expected content type {.field application/json}, not \\
-                    {.field <<x>>}."
-      ),
-      "*" = obfuscate(content, first = 197, last = 0)
-    ),
+    "Expected content type {.field application/json}, not {.field {type}}.",
     call = call,
     resp = resp
   )
 }
 
-# personal policy: a wrapper around a wrapper around cli_abort() should not
-# capture/pass an environment
-# if you really want cli styling, you have to pre-interpolate
-gargle_abort_request_failed <- function(message, resp, call = caller_env()) {
+gargle_abort_request_failed <- function(message,
+                                        resp,
+                                        .envir = parent.frame(),
+                                        call = caller_env()) {
   gargle_abort(
     message,
     class = c(
       "gargle_error_request_failed",
       glue("http_error_{httr::status_code(resp)}")
     ),
+    .envir = .envir,
     call = call,
     resp = redact_response(resp)
   )
 }
 
 #' @export
 #' @rdname response_process
-gargle_error_message <- function(resp) {
-  content <- response_as_json(resp)
+gargle_error_message <- function(resp, call = caller_env()) {
+  type <- httr::http_type(resp)
+  if (type == "text/html") {
+    return(gargle_html_error_message(resp))
```

Comment:
Instead of letting html trigger an error in `check_for_json()` inside `response_as_json()`, I handle that case explicitly here.